plugins/jobs/src/js/components/JobsOverviewTable.js

```diff
@@ -69,37 +68,68 @@ class JobsTable extends React.Component {
       }
     ];
   }
-
-  // TODO: DCOS-7766 Revisit this pre-rendering data transformation...
-  getData() {
-    return this.props.jobs.map(function(job) {
-      const isGroup = job instanceof Tree;
-      let lastRun = {};
-      let schedules = null;
-      let status = null;
-
-      if (!isGroup) {
-        const lastRunsSummary = job.getLastRunsSummary();
-
-        lastRun = {
-          status: job.getLastRunStatus().status,
-          lastSuccessAt: lastRunsSummary.lastSuccessAt,
-          lastFailureAt: lastRunsSummary.lastFailureAt
+  /**
+   * converts data from Job to table format.
+   *
+   * @param  {JobConnection} data
+   * @returns {Array<{ id, idGroup, name, schedules, status, lastRun }>} object with data needed by Table Component
+   */
+  getData(data) {
+    return Object.values(
+      data.nodes.reduce((acc, job) => {
```

Comment:
:art: I think given the complexity of what is being done here, I would extract the reduce function on its own and 

`return Object.values(data.nodes.reduce(nameYourReducer))`